implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v1/DynatraceExporterV1.java

```diff
@@ -279,4 +282,22 @@ private Meter.Id idWithSuffix(Meter.Id id, String suffix) {
         return id.withName(id.getName() + "." + suffix);
     }
 
+    /**
+     * Redacts the API token from a thrown exception before printing the exception
+     * message.
+     * @param t the original {@link Throwable}
+     * @return the original {@link Throwable} if it does not contain the API token or a
+     * new {@link Throwable} with the message with the token redacted and same stack trace
+     * as the original Throwable.
+     */
+    private Throwable redactToken(Throwable t) {
+        if (!t.getMessage().contains(config.apiToken())) {
+            return t;
+        }
+
+        Throwable throwable = new Throwable(t.getMessage().replace(config.apiToken(), "<redacted>"));
```

Comment:
Could we have a test covering this new handling?

Reply:
It was a pain to test, but I think it should be covered now for both put and post.